Add Docker Support

LDShadowLord opened the pull request:
- Added Dockerfile to build container. Based on Alpine 3.14, Nginx 1.2, and PHP8.
- Added GitHub actions to auto-build containers on tagged release.
- Added docker-conf folder containing pre-build config files
- Added docker-conf/empty folder for use in the Dockerfile build process.

LDShadowLord committed:
Added Config files for Docker

LDShadowLord committed:
Added Dockerfile and Entrypoint

LDShadowLord committed:
Added Docker Instructions for README

LDShadowLord committed:
Added Dockerfile for build

LDShadowLord committed:
Fixed Build Process

LDShadowLord committed:
Create docker-publish.yml

First Draft of Auto-Docker Built

LDShadowLord committed:
Added empty files to fix Docker Build Failure

LDShadowLord committed:
Merge branch 'master' of https://github.com/LDShadowLord/Booklib

LDShadowLord committed:
Fix Workflow?

LDShadowLord committed:
Updated README

LDShadowLord committed:
Fixed Workflows to properly tag releases

LDShadowLord committed:
Removed unnecessary files

LDShadowLord committed:
Fixed Permissions Issue.

LDShadowLord committed:
Are these files necessary?

LDShadowLord committed:
These files are unnecessary

LDShadowLord committed:
Fix [1] To Github Workflow

LDShadowLord committed:
Delete empty

LDShadowLord committed:
Merge branch 'master' of https://github.com/LDShadowLord/Booklib

LDShadowLord committed:
Added Build Cache

Should speed up builds in the future

LDShadowLord committed:
Fix Cache

LDShadowLord committed:
Added Fix for Cache exponential increase bug

LDShadowLord committed:
Added empty file so Github doesn't throw a fit

LDShadowLord committed:
Fixed docker to pull empty directory

LDShadowLord committed:
Fixed Caching Again

Final Release?

LDShadowLord committed:
Removed composer self-update & Changed Maintainer

LDShadowLord commented:
The README will have to be updated to point at `ghcr.io/MKaterbarg/Booklib` rather than `ghcr.io/LDShadowLord/Booklib`. Sorry about that, forgot.

MKaterbarg commented on init.sh:
```diff
@@ -0,0 +1,23 @@
+    touch $CONTAINER_READY
+    echo "Running first-time startup logic"
+    php artisan key:generate
+    php artisan migrate
```
migrate and db:seed can be run every time the container starts, since it won’t re-run already executed commands. After each upgrade, this should be run, so it may be worth adding this to the startup for each launch, as well as a git pull. Else building an upgrade function in the web-UI is an option. Will consider

LDShadowLord replied:
I ended up ditching this logic, due to a bug. I'll try and revisit this to get this working so I can automate the initial/upgrade experience.

LDShadowLord replied:
Added all of these features (except the web ui stuff, that's not my department).

LDShadowLord committed:
Update README.md

LDShadowLord committed:
Preparing Entrypoint

LDShadowLord committed:
Merge branch 'master' of https://github.com/LDShadowLord/Booklib

LDShadowLord committed:
Added initial logic

LDShadowLord committed:
Now pulls from my Github for testing.

LDShadowLord committed:
Added more verbose output

LDShadowLord committed:
Added all logic

LDShadowLord committed:
Removed unnecessary layers

LDShadowLord committed:
Updated Logic to keep container alive

LDShadowLord committed:
Dockerfile now launches python properly

LDShadowLord committed:
Updated README to remove extraneous instructions

LDShadowLord committed:
Merge branch 'MKaterbarg:master' into master

LDShadowLord committed:
Updated Init script to add "auto-update" function.

LDShadowLord committed:
Added proper logging + code cleaning

LDShadowLord committed:
Added auto_update environmental variable

LDShadowLord committed:
Fixed formatting.

LDShadowLord committed:
Changed my git to Martijns.

LDShadowLord commented:
Added startup script which invalidates the need for a .env file to be provided. Any entry in the .env file can now be passed an as environmental variable and the .env file will be built (on first startup ONLY). If the .env file is required to be changed, it needs to be manually deleted and the script will rebuild it from the environmental variables again or the file can be manually edited.

Script also runs the migrate and db:seed commands automatically on every container start, and runs key:generate on first time startup. First time startup is denoted by the file "Init_Success" being found in /storage/. If it doesn't find this file, it will run key:generate again.

Script will attempt a git pull UNLESS "AUTO_UPDATE=FALSE" is set in the environmental variables, in which case it will not attempt an auto-update (Some people want to stay on a static version until the version is battle tested).

Added logging, so everything the script does will be logged to /storage/logs/init.log in case of a failure.
